{{cookiecutter.profile_name}}/slurm-status.py

```diff
@@ -1,52 +1,86 @@
 #!/usr/bin/env python3
+import json
+import os
 import re
+import requests
 import subprocess as sp
 import shlex
 import sys
 import time
 import logging
 from CookieCutter import CookieCutter
 
-logger = logging.getLogger("__name__")
+logger = logging.getLogger(__name__)
 
 STATUS_ATTEMPTS = 20
+SIDECAR_VARS = os.environ.get("SNAKEMAKE_CLUSTER_SIDECAR_VARS", None)
+DEBUG = bool(int(os.environ.get("SNAKEMAKE_SLURM_DEBUG", "0")))
 
-jobid = sys.argv[1]
+if DEBUG:
+    logging.basicConfig(level=logging.DEBUG)
+    logger.setLevel(logging.DEBUG)
 
-cluster = CookieCutter.get_cluster_option()
 
-for i in range(STATUS_ATTEMPTS):
-    try:
-        sacct_res = sp.check_output(shlex.split(f"sacct {cluster} -P -b -j {jobid} -n"))
-        res = {
-            x.split("|")[0]: x.split("|")[1]
-            for x in sacct_res.decode().strip().split("\n")
-        }
-        break
-    except sp.CalledProcessError as e:
-        logger.error("sacct process error")
-        logger.error(e)
-    except IndexError as e:
-        logger.error(e)
-        pass
-    # Try getting job with scontrol instead in case sacct is misconfigured
+def get_status_direct(jobid):
+    """Get status directly from sacct/scontrol"""
+    cluster = CookieCutter.get_cluster_option()
+    for i in range(STATUS_ATTEMPTS):
+        try:
+            sacct_res = sp.check_output(shlex.split(f"sacct {cluster} -P -b -j {jobid} -n"))
+            res = {x.split("|")[0]: x.split("|")[1] for x in sacct_res.decode().strip().split("\n")}
+            break
+        except sp.CalledProcessError as e:
+            logger.error("sacct process error")
+            logger.error(e)
+        except IndexError as e:
+            logger.error(e)
+            pass
+        # Try getting job with scontrol instead in case sacct is misconfigured
+        try:
+            sctrl_res = sp.check_output(shlex.split(f"scontrol {cluster} -o show job {jobid}"))
+            m = re.search(r"JobState=(\w+)", sctrl_res.decode())
+            res = {jobid: m.group(1)}
+            break
+        except sp.CalledProcessError as e:
+            logger.error("scontrol process error")
+            logger.error(e)
+            if i >= STATUS_ATTEMPTS - 1:
+                print("failed")
+                exit(0)
+            else:
+                time.sleep(1)
+
+        return res[jobid] or ""
```

Comment:
Dedent the return statement one level, otherwise get_status_direct returns None

Reply:
Can do.